File: frontend/src/backend/dailyPanelService.js
import {config, appFetch } from './appFetch';


export const addSaleRoom = (idRoomType, date, freeRooms, onSuccess, onErrors) => {

    //const date = saleRoomDate.getTime();

    appFetch('/dailyPanel/addSaleRoom', config('POST', {idRoomType, date, freeRooms}), 
    onSuccess, onErrors);
}


export const uploadSaleRoomTariff = (price, tariffId, roomTypeId, date, onSuccess, onErrors) => {
  
    //const date = saleRoomTariffDate.getTime();

    appFetch('/dailyPanel/uploadSaleRoomTariff', config('POST', {price, tariffId, roomTypeId, date}), 
    onSuccess, onErrors);
}


export const findDailyPanel = (date, onSuccess, onErrors) => {

    let path = `/dailyPanel/findDailyPanel?date=${date}`;

    appFetch(path, config('GET'), onSuccess, onErrors);
}

File: backend/src/main/java/es/udc/tfg/backend/model/services/DailyPanelService.java
package es.udc.tfg.backend.model.services;

import java.math.BigDecimal;
import java.util.Calendar;
import java.util.List;

import es.udc.tfg.backend.model.common.exceptions.DuplicateInstanceException;
import es.udc.tfg.backend.model.common.exceptions.InstanceNotFoundException;
import es.udc.tfg.backend.model.entities.RoomTable;
import es.udc.tfg.backend.model.entities.SaleRoom;
import es.udc.tfg.backend.model.entities.SaleRoomTariff;

public interface DailyPanelService {

	SaleRoomTariff uploadSaleRoomTariff(BigDecimal price, Long tariffId, Long roomTypeId, Calendar date) throws InstanceNotFoundException, PriceNotBetweenMinAndMaxValueException;

	SaleRoom addSaleRoom(Long roomTypeId, Calendar date, int freeRooms) throws DuplicateInstanceException, InstanceNotFoundException, FreeRoomsLessThanRoomTypeQuantityException;
	
	List<RoomTable> findDailyPanel (Calendar initialDate);
	
}